### PR TITLE
Maxpool ceilmode

### PR DESCRIPTION
**Pull request by owulveryck**
**TL;DR:** This PR is a draft to open the discussion about a possibility to break the API for the Maxpool operator to add a "ceil mode".

**Longer:** 

From version 1.5, onnx supports a "ceil mode" attribute that has an impact on the calculus of the output shape on the Maxpool operator. (see [this disccusion](https://github.com/onnx/onnx/issues/549) for more info).

By now, I did not find any model that is using this mode, nevertheless, I've implemented a basic implementation to do different tests (one again **I am only using this branch for testing purpose**.)

The problem is that it breaks the API by adding a new attribute to the `MaxPool2D` function; another option would be to keep this function as-is and silently set the `ceilMode` to false and to create a new transition function `MaxPool2DCeil` but it do not really like this idea.

WDYT?

### CODE REVIEW

**Comment by owulveryck** [comment]
One more thing: this branch also includes in shadow the premises of implementation of `dilation`

**Comment by chewxy** [comment]
I like it. Perhaps we should think of using `...interface{}` for `MaxPool2D` for now. This allows for a future transition to using functional options like 

```
type NNOpt func(Op) error
```

which would fit into the building of golgi nicely in the future

**Comment by zimenglan-sysu-512** [comment]
any progress updates?
